src/cleanup.ts

```diff
@@ -14,9 +19,59 @@ export async function run(cutoff = '24h'): Promise<void> {
     await docker.pruneNetworks({filters: untilFilter})
     core.info(`Pruning containers older than ${cutoff}`)
     await docker.pruneContainers({filters: untilFilter})
+    await cleanupOldImageVersions(docker, UPDATER_IMAGE_NAME)
+    await cleanupOldImageVersions(docker, PROXY_IMAGE_NAME)
   } catch (error) {
     core.error(`Error cleaning up: ${error.message}`)
   }
 }
 
+export async function cleanupOldImageVersions(
+  docker: Docker,
+  imageName: string
+): Promise<void> {
+  const repo = repositoryName(imageName)
+  const options = {
+    filters: {
+      reference: [repo]
+    }
+  }
+
+  core.info(`Cleaning up images for ${repo}`)
+
+  docker.listImages(options, async function (err, images) {
+    if (images && images.length > 0) {
+      for (const imageInfo of images) {
+        // The given imageName is expected to be a digest, however to avoid any surprises in future
+        // we fail over to check for a match on tags as well.
+        //
+        // This means we won't remove any image which matches an imageName of either of these notations:
+        // - dependabot/image@sha256:$REF (current implementation)
+        // - dependabot/image:v1
+        //
+        // Without checking imageInfo.RepoTags for a match, we would actually remove the latter even if
+        // this was the active version.
+        if (
+          imageInfo.RepoDigests?.includes(imageName) ||
+          imageInfo.RepoTags?.includes(imageName)
```

Comment:
It wasn't immediately obvious to me where `imageInfo` came from. I don't think I was expecting it to be a member of the collection `images`. I'm guessing the intent here was to avoid using `image` as it's so close to `images`.

💡 I think that `image.RepoDigests?.includes` reads pretty naturally. This has me wondering if we could use another name for the collection. The function name is `cleanupOldImageVersions`. What do you think of the name `oldImages`?

```typescript
  docker.listImages(options, async function (err, oldImages) {
     if (oldImages && oldImages.length > 0) {
        for (const image of oldImages) {
```

If this doesn't resonate for you, please feel free to acknowledge with a 👀 reaction and resolve 😄

Reply:
Could use `imageInfo of imageInfoList` ?